koku/api/utils.py

```diff
@@ -252,6 +268,28 @@ def n_days_ago(self, in_date, n_days):
         n_days = midnight - datetime.timedelta(days=n_days)
         return n_days
 
+    def n_days_ago_or_month_start(self, in_date, n_days):
```

Comment:
Question: 

I'm curious why we need this new `n_days_ago_or_month_start` method that could not be addressed with the existing methods `n_days_ago` and  `month_start` or `month_start_utc`?

Reply:
So this is using the n_days_ago under the hood, and we could rely on both of them outside of this, we would just need _some_ logic inside the validation class to make the correct lookup. Happy to change it though.